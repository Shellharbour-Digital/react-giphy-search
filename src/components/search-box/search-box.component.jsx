import React from 'react';

import { SearchInput, SearchContainer } from './search-box.styles';

export const SearchBox = ({ placeholder, handleChange, debounceTimeout, ...otherProps }) => (
  <SearchContainer>
    <SearchInput
      type='search'
      placeholder={placeholder}
      onChange={handleChange}
      debounceTimeout={debounceTimeout}
    />
  </SearchContainer>
);
