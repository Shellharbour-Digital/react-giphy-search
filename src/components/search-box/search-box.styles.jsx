import styled from 'styled-components';

import { DebounceInput } from 'react-debounce-input';

export const SearchInput = styled(DebounceInput)`
  display: block;
  margin: 0 auto;
  text-align: center;
  width: 40%;
  height: 40px;
  font-size: 18px;

  @media screen and (max-width: 800px) {
    width: 75%;
    height: 30px;
    font-size: 16px;
  }
`;

export const SearchContainer = styled.div`
  position: fixed;
  width: 100%;
  background-color: black;
  padding: 25px 0;
  z-index: 1;

  @media screen and (max-width: 800px) {
    padding: 15px 0;
  }
`;
