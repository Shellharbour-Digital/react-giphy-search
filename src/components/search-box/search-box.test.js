import { shallow } from 'enzyme';
import React from 'react';
import {SearchBox} from './search-box.component';

describe('SearchBox component', () => {
  let wrapper = shallow(<SearchBox />);

  it('expect to render SearchBox component', ()  => {
    expect(wrapper).toMatchSnapshot();
  });
});
