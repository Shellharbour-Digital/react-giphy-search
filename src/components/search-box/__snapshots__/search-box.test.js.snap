// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component expect to render SearchBox component 1`] = `
<styled.div>
  <Styled(DebounceInput)
    type="search"
  />
</styled.div>
`;
